### PR TITLE
8299375: [PPC64] GetStackTraceSuspendedStressTest tries to deoptimize frame with invalid fp

### PR DESCRIPTION
**Pull request by reinrich**
Mark a frame as not fully initialized when copying it from a continuation StackChunk to the stack until the callers_sp (aka back link) is set.

This avoids the assertion given in the bug report when the copied frame is deoptimized before it is fully initialized.
IMHO the deoptimization at that point is a little questionable but it actually only changes the pc of the frame which can be done.
Note that the frame can get extended later (and metadata can get overridden) but [there is code that handles this](https://github.com/openjdk/jdk/blob/34a92466a615415b76c8cb6010ff7e6e1a1d63b4/src/hotspot/share/runtime/continuationFreezeThaw.cpp#L2108-L2110).

Testing: jdk_loom. The fix passed our CI testing. This includes most JCK and JTREG tiers 1-4, also in Xcomp mode, on the standard platforms and also on ppc64le.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8299375](https://bugs.openjdk.org/browse/JDK-8299375): [PPC64] GetStackTraceSuspendedStressTest tries to deoptimize frame with invalid fp

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12941/head:pull/12941` \
`$ git checkout pull/12941`

Update a local copy of the PR: \
`$ git checkout pull/12941` \
`$ git pull https://git.openjdk.org/jdk pull/12941/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12941`

View PR using the GUI difftool: \
`$ git pr show -t 12941`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12941.diff">https://git.openjdk.org/jdk/pull/12941.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/ppc/frame_ppc.hpp [hunk 1]
```diff
@@ -379,6 +386,9 @@
   int offset_fp() const         { assert_offset();  return _offset_fp; }
   void set_offset_fp(int value) { assert_on_heap(); _offset_fp = value; }
 
+  // Mark a frame as not fully initialized
```
Comment (TheRealMDoerr) [inline src/hotspot/cpu/ppc/frame_ppc.hpp:389]: Maybe add a comment like "Must not be used for frames in the valid back chain."?

**Review by TheRealMDoerr — COMMENTED** [review]
This looks good. Seems like we should update Copyright years.

File: src/hotspot/cpu/ppc/frame_ppc.hpp (hunk 1 above)
Comment (reinrich) [reply src/hotspot/cpu/ppc/frame_ppc.hpp]: Thanks for the feedback. I've added the comment and updated the Copyright years.

**Review by TheRealMDoerr — APPROVED** [review]
Thanks!

**Pull request closed without merging**